signature-specification.md

```diff
@@ -0,0 +1,56 @@
+A signature envelope ![\varepsilon_n = \{ m, (\mu_0, \hat\mu_0, s_0), \dots, (\mu_{n-1}, \hat\mu_{n-1}, s_{n-1}) \}](https://render.githubusercontent.com/render/math?math=%5Ctextstyle+%5Cvarepsilon_n+%3D+%5C%7B+m%2C+%28%5Cmu_0%2C+%5Chat%5Cmu_0%2C+s_0%29%2C+%5Cdots%2C+%28%5Cmu_%7Bn-1%7D%2C+%5Chat%5Cmu_%7Bn-1%7D%2C+s_%7Bn-1%7D%29+%5C%7D) is a compact form of `n` signatures signing the same payload `m`.
+
+![A complete signature in OCI registry](media/signature-structure.svg)
+
+
+## Requirements
+
+### Signature Envelope and Signature
+
+1. The signature envelope payload MUST contain a [descriptor](https://github.com/opencontainers/image-spec/blob/master/descriptor.md#properties) of the target manifest.
+    1. A descriptor MUST contain `mediaType`, `digest`, `size` fields.
+    2. A descriptor MAY contain the `artifactType` field for artifact manifests, or the `config.mediaType` for `oci.image` based manifests.
+    3. descriptor MAY contain a `annotations` field.
+2. The signature envelope MUST be self contained to validate the integrity of the payload and signed attributes.
+3. The signature envelope MAY contain arbitrary user supplied information, as key value pairs. If included, it MUST be integrity protected.
+4. The signature envelope MAY contain information to determine the signature expiration time. If included, it MUST be integrity protected.
+5. The signature envelope MAY contain information to determine the signature revocation status. If included, it MUST be integrity protected.
+6. The signature MUST be detached a signature, with respect to the target manifest being signed.
+7.  The signature MUST be stored and transferred in a signature envelope.
+8.  If the signature was produced by a X.509 certificate, the signature MUST contain a certificate chain of the signing certificate as an unsigned attribute. This MAY be a partial certificate chain.
+9.  The signature MAY be timestamped by TSA, and this counter signature MUST be stored as an unsigned attribute.
+
+### Signature Format
+
+1. The signature format MUST be able to store payload.
+2. The signature format MUST be able to store cryptographic signature.
+3. The signature format MUST support referencing and in-lining of different signing identity types represented as keyIds or X.509 certificate chains.
+4. The signature format MUST be able store unsigned signature attributes such as timestamp signature (TSA).
+5. Signature format SHOULD be easy to implement using existing Go libraries (Docker CLI is implemented in Go).
+6. The signature format MAY be able store signed attributes. In case signature format doesn't support signed attributes, they can be combined with the payload before signing. Since the payload contains the signature specific metadata, the signature format loses the ability to store multiple signatures.

```

Comment:
My preference is we assume the payload include all attributes being signed since that gives us flexibility to use different signature formats.